include/alpaka/test/Check.hpp

```diff
@@ -9,28 +9,46 @@
             {                                                                                                         \
-                acc.cout << "ALPAKA_CHECK failed because '!(" << #expression << ")'\n";                               \
+                acc.cout << "ALPAKA_CHECK failed because '!(" << #expression << ")' in " << file << ":" << line       \
+                         << "\n";                                                                                     \
                 success = false;                                                                                      \
             }                                                                                                         \
         } while(0)
 #else
-#    define ALPAKA_CHECK(success, expression)                                                                         \
+#    define ALPAKA_CHECK_DO(printMsg, file, line, success, expression)                                                \
         do                                                                                                            \
         {                                                                                                             \
             if(!(expression))                                                                                         \
             {                                                                                                         \
-                printf("ALPAKA_CHECK failed because '!(%s)'\n", #expression);                                         \
+                printf("ALPAKA_CHECK failed because '!(%s)' in %s:%d\n", #expression, file, line);                    \
                 success = false;                                                                                      \
             }                                                                                                         \
         } while(0)
 #endif
+
+#if BOOST_LANG_HIP == BOOST_VERSION_NUMBER(4, 5, 0)
+// disable message print to avoid compiler error: 'error: stack size limit exceeded (181896) in _ZN6alpaka...'
+#    define ALPAKA_CHECK(printMsg, file, line, success, expression)                                                   \
+        do                                                                                                            \
+        {                                                                                                             \
+            if(!(expression))                                                                                         \
+            {                                                                                                         \
+                printf("ALPAKA_CHECK failed because '!(%s)' in %s:%d\n", #expression, file, line);                    \
```

Comment:
Do we need special handling for SYCL here? We don't have device side `printf` there.

Reply:
We do have it now.